vta/hardware/chisel/src/test/scala/ComputeTests.scala

```diff
@@ -0,0 +1,913 @@
+                           biases2_data4, biases2_data5, biases2_data6, biases2_data7)
+
+  // max
+  val biases3_data0 = "0000001a0000000b0000003600000004ffffffdf00000035fffffff200000003ffffffd4ffffffe1ffffffdcfffffffeffffffc300000028000000380000002e"
+  val biases3_data1 = "ffffffd8ffffffe5ffffffc9ffffffc5ffffffd6ffffffd000000033ffffffd00000003afffffff5ffffffd6ffffffe20000003bffffffc50000000900000014"
+  val biases3_data2 = "fffffff9ffffffe60000001c00000017ffffffdaffffffed0000001b0000000fffffffc3000000210000002a00000035000000190000000dfffffff4ffffffc2"
+  val biases3_data3 = "00000010fffffffc000000070000000fffffffcd00000009ffffffd4ffffffe600000021fffffff2ffffffd20000002cffffffdc0000001effffffc3fffffffc"
+  val biases3_data4 = "ffffffecffffffd30000002400000003ffffffd0ffffffe1ffffffc500000038ffffffc500000037000000060000003bffffffc7ffffffd4ffffffffffffffc7"
+  val biases3_data5 = "00000020ffffffd9ffffffcb00000007ffffffc4fffffff2000000190000002bffffffdffffffffcffffffc100000018ffffffdaffffffcdfffffff1ffffffea"
+  val biases3_data6 = "fffffffc0000003900000016ffffffefffffffcafffffff90000003800000024ffffffe00000000d0000003b00000038ffffffd7ffffffe500000021ffffffe7"
+  val biases3_data7 = "00000010ffffffe3ffffffddffffffd3ffffffc2fffffff3ffffffcbffffffc8fffffffafffffff9ffffffedffffffcf0000003e0000003bffffffd900000005"
+  val biases3 = IndexedSeq(biases3_data0, biases3_data1, biases3_data2, biases3_data3,
+                           biases3_data4, biases3_data5, biases3_data6, biases3_data7)
+
+  // shr
+  val biases4_data0 = "ffffffd6ffffffc1ffffffd6ffffffeaffffffdfffffffdbffffffc50000001000000037fffffffa000000020000002100000000006646c000007ffff6cd4878"
+  val biases4_data1 = "00000019ffffffc1ffffffe4ffffffc5fffffff700000012ffffffc600000030fffffffd00000033ffffffc5ffffffc400000012ffffffe90000001fffffffc4"
+  val biases4_data2 = "0000003bffffffd7ffffffc4ffffffd800000007ffffffeafffffff100000019ffffffc2ffffffd5ffffffd5ffffffdd0000003e00000035ffffffc4ffffffc8"
+  val biases4_data3 = "0000000d0000003100000010fffffff900000031000000200000002f00000030fffffff50000002bffffffecffffffe0ffffffdb00000008fffffff0ffffffcf"
+  val biases4_data4 = "000000200000003b000000210000003cffffffd5000000310000002a0000001affffffcaffffffc30000000300000004ffffffea00000000ffffffeb0000001f"
+  val biases4_data5 = "ffffffc5000000220000000efffffff700000016ffffffdbffffffc90000003c00000030ffffffeaffffffd30000003c0000002dfffffffa0000003f0000003d"
+  val biases4_data6 = "0000000dffffffd4000000330000003dffffffd300000019000000370000003bffffffd80000002dffffffd50000003e000000230000001200000034ffffffed"
+  val biases4_data7 = "ffffffcafffffffd00000026ffffffe50000002e0000002affffffdcfffffffc00000035ffffffd9ffffffc1fffffffe0000001000000026ffffffd400000034"
+  val biases4 = IndexedSeq(biases4_data0, biases4_data1, biases4_data2, biases4_data3,
+                           biases4_data4, biases4_data5, biases4_data6, biases4_data7)
+
+  def init_compute0() {
```

Comment:
if you're not averse to using bleeding edge repos, [chisel-testers2](https://github.com/ucb-bar/chisel-testers2) has a nice interface for testing `DecoupledIO`